appengine/src/main/java/com/google/googleinterns/gscribe/modules/ServicesModule.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.googleinterns.gscribe.modules;
+
+import com.google.googleinterns.gscribe.services.ExamParserService;
+import com.google.googleinterns.gscribe.services.TokenService;
+import com.google.googleinterns.gscribe.services.impl.ExamParserServiceImpl;
+import com.google.googleinterns.gscribe.services.impl.TokenServiceImpl;
+import com.google.googleinterns.gscribe.utils.*;
+import com.google.inject.AbstractModule;
+import com.google.inject.Inject;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+
+public class ServicesModule extends AbstractModule {
+
+    @Inject
+    @Provides
+    @Singleton
+    public ExamParserService examParserServiceProvider(ExamGenerator examGenerator, ExamValidator examValidator, GetExamFromSpreadsheet getExamFromSpreadsheet) {
```

Comment:
There is also a binding above, make sure it is getting the binding from the right place. Remove either, and make sure that both the Modules are added when the Injector object is created in App.